libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -27,36 +28,111 @@ CGroupsParser::CGroupsParser()
+    removeUnit.pop_back();
+
+    if (std::isdigit(suffix)) {
+        //the value is digit already no need to do anything
+        return settingValue;
+    } else if ('K' == suffix || 'k' == suffix) {
+        //the value is Kilobyte unit convert it to byte
+        int KB = 1024;
+
+        try {
+            return byteRegulation(removeUnit, KB);
+        } catch (LimitRangeError &error) {
+            throw LimitRangeError();
+        }
+    } else if ('M' == suffix || 'm' == suffix) {
+        //the value is Megabyte unit convert it to byte
+        int MB = 1024 * 1024;
+
+        try {
+            return byteRegulation(removeUnit, MB);
+        } catch (LimitRangeError &error) {
+            throw LimitRangeError();
+        }
+    } else if ('G' == suffix || 'g' == suffix) {
+        //the value is Gigabyte unit convert it to byte
+        int GB = 1024 * 1024 * 1024;
+
+        try {
+            return byteRegulation(removeUnit, GB);
+        } catch (LimitRangeError &error) {
+            throw LimitRangeError();
+        }
+    }
+    log_error() << "Bad suffix on setting value " << settingValue;
+    throw BadSuffixError("Bad suffix on setting value " + settingValue);
+}
+
+void CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
 {
     std::string settingKey;
     std::string settingValue;
 
     if (!JSONParser::read(element, "setting", settingKey)) {
         log_error() << "Key \"setting\" either not a string or not in json configuration";
-        return ReturnCode::FAILURE;
+        throw JSonError();
     }
 
     if (!JSONParser::read(element, "value", settingValue)) {
         log_error() << "Key \"value\" either not a string or not in json configuration";
-        return ReturnCode::FAILURE;
+        throw JSonError();
     }
 
-    if ("memory.limit_in_bytes" == settingKey) {
+    if (("memory.limit_in_bytes" == settingKey) || ("memory.memsw.limit_in_bytes" == settingKey)) {
         if (m_settings.count(settingKey)) {
+            try {
+                settingValue = suffixCorrection(settingValue);
+            } catch (LimitRangeError &error) {
+                log_error() << settingValue << " is not appropriate value for " << settingKey;
+                throw LimitRangeError();
+            } catch (BadSuffixError  &error) {
+                throw BadSuffixError();
+            }
+
             // if the new value is greater than old one set is as the value
-            if (std::stoi(m_settings[settingKey]) < std::stoi(settingValue)) {
+            if (std::stoll(m_settings[settingKey]) < std::stoll(settingValue)) {
                 m_settings[settingKey] = settingValue;
             }
         } else {
+            try {
+                settingValue = suffixCorrection(settingValue);
+            } catch (LimitRangeError &error) {
+                log_error() << settingValue << " is not appropriate value for " << settingKey;
+                throw LimitRangeError();
```

Comment:
Seems like this code is identical to the code in the other case of the if statement or at least very similar. This can be cleaned up